vm/src/value.rs

```diff
@@ -439,7 +439,23 @@ impl<'a, 't> InternalPrinter<'a, 't> {
             Value::Userdata(ref data) => arena.text(format!("{:?}", data)),
             Value::Thread(thread) => arena.text(format!("{:?}", thread)),
             Value::Byte(b) => arena.text(format!("{}", b)),
-            Value::Int(i) => arena.text(format!("{}", i)),
+            Value::Int(i) => {
+                use base::types::BuiltinType;
+                match **self.typ {
+                    Type::Builtin(BuiltinType::Int) => arena.text(format!("{}", i)),
+                    Type::Builtin(BuiltinType::Char) =>
+                        if 0 <= i && i <= ::std::u32::MAX as isize {
```

Comment:
I think it should be fine to just omit this comparison. When `isize` is 32-bit I think `i as u32` should should preserve all the bits (that is, `i as i32 as u32 == i`).

Reply:
When `isize` is 64-bit, how can I deal with it?
Should I forcibly cast i to u32, then call `char::from_u32` (that is, `char::from_u32(i as u32)`)?

Reply:
Yep. Just truncating with `as` is fine.